### PR TITLE
Issue #5 destroy exercises

### LINKED ISSUE #5: Destroy exercises
Guide edition should allow to remove exercises from guide. 

<del>Special attention should be put on also removing exercise from git repository</del>

### PR DESCRIPTION
**Pull request by aguspina**
Fixes #5

### CODE REVIEW

File: app/src/editor/ExerciseController.js [hunk 1]
```diff
@@ -70,6 +78,34 @@
...
+          $scope.publish();
+        }
+      })
+
+    $scope.delete = function() {
+      var toast = $mdToast.simple()
+            .textContent('Exercise deleted!')
+            .action('undo')
+            .highlightAction(true);
+      $mdToast.show(toast).then(function(response) {
```
Comment (faloi) [inline app/src/editor/ExerciseController.js]: This is quite the same thing that you do for guides. 

You can create a custom service that receives the text (`'Exercise deleted'`) and the action (lines 101-104) and shows the toast for you.

**Comment by aguspina** [comment]
@faloi done!

File: app/src/editor/DeleteService.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  angular.module('editor')
+         .service('deleteService', [
+          'store',
+          'navigationService',
+          '$mdToast',
+          DeleteService
+         ]);
+
+  function DeleteService(store, navigationService, $mdToast){
```
Comment (faloi) [inline app/src/editor/DeleteService.js]: Great, but this has nothing to do with _delete_. It's just a service that shows a toast for "undoable" things. Also, the `navigationService` is not used.

Comment (aguspina) [inline app/src/editor/DeleteService.js]: You are right, i'm stupid today

Comment (faloi) [inline app/src/editor/DeleteService.js]: Thinking it a little more, this sounds more like a directive (name is quite awful, I know):

```jade
undoable-button(action="delete()", undo-message="Exercise deleted!")
```

File: app/index.html [hunk 3]
```diff
@@ -65,6 +65,7 @@
     <script src="./src/editor/ExerciseController.js"></script>
     <script src="./src/editor/NavigationService.js"></script>
     <script src="./src/editor/SessionService.js"></script>
+    <script src="./src/editor/DeleteService.js"></script>
```
Comment (faloi) [inline app/index.html]: This doesn't seem right.

File: app/src/editor/UndoableService.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  'use strict';
+
+  angular.module('editor')
+         .service('undoableService', [
+          'store',
+          '$mdToast',
+          UndoableService
+         ]);
+
+  function UndoableService(store, $mdToast){
```
Comment (faloi) [inline app/src/editor/UndoableService.js]: Thinking it a little more, this sounds more like a directive (name is quite awful, I know):

```jade
undoable-button(action="delete()", undo-message="Exercise deleted!")
```

But for now the current approach is fine.

**Pull request merged**